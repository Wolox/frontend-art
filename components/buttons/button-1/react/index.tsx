import React from 'react';

import '../styles.scss';

function Button1() {
  return (
    <div className="column center btn-group-1">
      <button className="btn btn-1 m-bottom-2" disabled={true}>
        <div className="btn__content">cancel</div>
      </button>
      <button type="button" className="btn btn-1">
        <div className="btn__content">submit</div>
      </button>
    </div>
  );
}

export default Button1;
