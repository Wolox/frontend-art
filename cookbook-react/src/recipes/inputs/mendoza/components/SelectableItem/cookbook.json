{
  "thumbnail": {
    "type": "iframe",
    "url": "https://cookbook-react.now.sh/?category=inputs&component=mendoza&compact=true"
  },
  "detail": {
    "type": "iframe",
    "url": "https://cookbook-react.now.sh/?category=inputs&component=mendoza&compact=true"
  }
}
