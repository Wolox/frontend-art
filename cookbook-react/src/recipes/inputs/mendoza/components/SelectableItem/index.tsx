import React from 'react';
import cn from 'classnames';

import styles from './styles.module.scss';

interface ItemType {
  id: string;
  title: string;
  subtitle: string;
  price: number;
}

interface Props {
  item: ItemType;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  type: 'checkbox' | 'radio';
  selected: boolean;
}

function SelectableItem({ item, onChange, type, selected }: Props) {
  return (
    <label htmlFor={item.id} className={cn(styles.selectableItemContainer, { [styles.selected]: selected })}>
      <input
        id={item.id}
        className={styles.input}
        key={item.id}
        type={type}
        name={item.id}
        checked={selected}
        value={`${selected}`}
        onChange={onChange}
      />
      <div
        className={cn(styles.customCheckmark, {
          [styles.customCheckbox]: type === 'checkbox',
          [styles.customRadio]: type === 'radio'
        })}
      />
      <div className={styles.text}>
        <label htmlFor={item.id} className={styles.title}>
          {item.title}
        </label>
        <h6 className={styles.subtitle}>{item.subtitle}</h6>
      </div>
      <span className={styles.price}>{`$${item.price}`}</span>
    </label>
  );
}

export default SelectableItem;
