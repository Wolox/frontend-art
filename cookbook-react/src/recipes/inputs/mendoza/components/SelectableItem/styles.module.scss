$jumbo: #848486;
$alto: #DDD;
$silver: #CCC;
$international-klein-blue: #002EC9;

$selectable-size: 20px;

$radio-inner-size: 14px;
$radio-inner-translation: calc(50% - (#{$selectable-size} - #{$radio-inner-size}));

$checkbox-inner-size: 18px;

.custom-checkmark {
  border: 2px solid $alto;
  height: $selectable-size;
  margin-right: 10px;
  position: relative;
  width: $selectable-size;

  &::after {
    content: '';
    opacity: 0;
    transition: opacity 150ms ease-in-out;
  }
}

.custom-radio {
  border-radius: 50%;

  &::after {
    background-color: $international-klein-blue;
    border-radius: 50%;
    height: $radio-inner-size;
    position: absolute;
    transform: translate($radio-inner-translation, $radio-inner-translation);
    width: $radio-inner-size;
  }
}

.selectable-item-container {
  align-items: center;
  border: 1px solid $jumbo;
  border-radius: 7px;
  cursor: pointer;
  display: flex;
  height: 60px;
  justify-content: space-between;
  padding: 10px;
  position: relative;
  width: 250px;

  &.selected {
    border: 2px solid $international-klein-blue;
    padding: 9px;
    transition: border-color 150ms ease-in-out;
  }

  &:hover .input ~ .custom-checkmark {
    background-color: $silver;
  }

  &:hover .input[value='true'] ~ .custom-radio {
    background-color: $international-klein-blue;
  }

  .input[value='true'] ~ .custom-checkmark::after {
    opacity: 1;
  }
}

.text {
  display: flex;
  flex: 1;
  flex-direction: column;
  height: 100%;
  justify-content: space-between;
}

.input {
  cursor: pointer;
  height: 0;
  opacity: 0;
  position: absolute;
  width: 0;
}

.custom-checkbox {
  &::after {
    background: url('./assets/ic-check.svg') center no-repeat;
    height: $checkbox-inner-size;
    transform: translate(-1px, -1px);
    position: absolute;
    width: $checkbox-inner-size;
  }
}

.title {
  font-weight: 700;
}
