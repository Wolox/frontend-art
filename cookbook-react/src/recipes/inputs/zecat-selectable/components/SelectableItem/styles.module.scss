$jumbo: #848486;
$alto: #DDDDDD;

$selectable-size: 20px;

$radio-inner-size: 14px;
$radio-inner-translation: calc(50% - (#{$selectable-size} - #{$radio-inner-size}));

$checkbox-inner-size: 18px;

.selectable-item-container {
  align-items: center;
  border: 1px solid $jumbo;
  border-radius: 7px;
  cursor: pointer;
  display: flex;
  height: 60px;
  justify-content: space-between;
  padding: 10px;
  position: relative;
  width: 250px;

  &:hover .input ~ .custom-checkmark {
    background-color: #ccc;
  }

  &:hover .input[value="true"] ~ .custom-radio {
    background-color: #2196F3;
  }

  .input[value="true"] ~ .custom-checkmark:after {
    opacity: 1;
  }
}

.text {
  display: flex;
  flex: 1;
  flex-direction: column;
  height: 100%;
  justify-content: space-between;
}

.input {
  cursor: pointer;
  height: 0;
  opacity: 0;
  position: absolute;
  width: 0;
}

.custom-checkmark {
  border: 2px solid $alto;
  height: $selectable-size;
  margin-right: 10px;
  position: relative;
  width: $selectable-size;

  &:after {
    content: '';
    opacity: 0;
    transition: opacity 150ms ease-in-out;
  }
}

.custom-radio {
  border-radius: 50%;

  &:after {
    background-color: #2196F3;
    border-radius: 50%;
    height: $radio-inner-size;
    position: absolute;
    transform: translate($radio-inner-translation, $radio-inner-translation);
    width: $radio-inner-size;
  }
}

.custom-checkbox {
  &:after {
    background: url('./assets/ic-check.svg') center no-repeat;
    height: $checkbox-inner-size;
    transform: translate(-1px, -1px);
    position: absolute;
    width: $checkbox-inner-size;
  }
}

.title {
  font-weight: 700;
}
