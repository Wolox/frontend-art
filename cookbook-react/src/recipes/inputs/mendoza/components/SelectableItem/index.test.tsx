import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import SelectableItem from '.';

const item = { id: 'first_item', title: 'First item', subtitle: 'Small', price: 100 };

describe('when the selectable item is a radio', () => {
  test('Should match snapshot', () => {
    const { container } = render(<SelectableItem onChange={jest.fn()} selected={false} item={item} type="radio" />);
    expect(container).toMatchSnapshot();
  });
});

describe('when the selectable item is a checkbox', () => {
  test('Should match snapshot', () => {
    const { container } = render(<SelectableItem onChange={jest.fn()} selected={false} item={item} type="checkbox" />);
    expect(container).toMatchSnapshot();
  });
});

describe('when user interacts with the element', () => {
  const handleChange = jest.fn();

  beforeEach(() => {
    render(<SelectableItem onChange={handleChange} selected={false} item={item} type="checkbox" />);
  });

  test('When user selects the element the callback is called', () => {
    const selectableItem = screen.getByLabelText('First item') as HTMLInputElement;
    userEvent.click(selectableItem);
    expect(handleChange).toHaveBeenCalled();
  });
});

describe('when item is is not selected', () => {
  beforeEach(() => {
    render(<SelectableItem onChange={jest.fn()} selected={false} item={item} type="checkbox" />);
  });

  test('When user selects the element the callback is called', () => {
    const selectableItem = screen.getByLabelText('First item') as HTMLInputElement;
    expect(selectableItem.checked).toBe(false);
  });
});

describe('when item is is not selected', () => {
  beforeEach(() => {
    render(<SelectableItem onChange={jest.fn()} selected item={item} type="checkbox" />);
  });

  test('When user selects the element the callback is called', () => {
    const selectableItem = screen.getByLabelText('First item') as HTMLInputElement;
    expect(selectableItem.checked).toBe(true);
  });
});
