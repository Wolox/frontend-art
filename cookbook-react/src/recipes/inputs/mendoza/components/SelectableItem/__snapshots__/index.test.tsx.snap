// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the selectable item is a checkbox Should match snapshot 1`] = `
<div>
  <label
    class="selectableItemContainer"
    for="first_item"
  >
    <input
      class="input"
      id="first_item"
      name="first_item"
      type="checkbox"
      value="false"
    />
    <div
      class="customCheckmark customCheckbox"
    />
    <div
      class="text"
    >
      <label
        class="title"
        for="first_item"
      >
        First item
      </label>
      <h6
        class="subtitle"
      >
        Small
      </h6>
    </div>
    <span
      class="price"
    >
      $100
    </span>
  </label>
</div>
`;

exports[`when the selectable item is a radio Should match snapshot 1`] = `
<div>
  <label
    class="selectableItemContainer"
    for="first_item"
  >
    <input
      class="input"
      id="first_item"
      name="first_item"
      type="radio"
      value="false"
    />
    <div
      class="customCheckmark customRadio"
    />
    <div
      class="text"
    >
      <label
        class="title"
        for="first_item"
      >
        First item
      </label>
      <h6
        class="subtitle"
      >
        Small
      </h6>
    </div>
    <span
      class="price"
    >
      $100
    </span>
  </label>
</div>
`;
